sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -733,7 +733,7 @@ object SQLConf {
     .doc("When false, we will throw an error if a query contains a cartesian product without " +
         "explicit CROSS JOIN syntax.")
     .booleanConf
-    .createWithDefault(false)
+    .createWithDefault(true)
```

Comment:
can we also make this config internal? We know that there are problems in the cross join check and we may want to completely get rid of it at some point.